Git/commit.sublime-syntax

```diff
@@ -0,0 +1,81 @@
+%YAML 1.2
+---
+# Highlight regular git commits, merge commits, and tags.
+
+name: Git Commit
+file_extensions:
+  - COMMIT_EDITMSG
+  - MERGE_MSG
+  - TAG_EDITMSG
+scope: source.gitcommit
+
+contexts:
+  prototype:
+    - include: comments
+
+  main:
+    - include: commit-message
+
+  branch-line:
+    - match: \bOn branch\s+(.*)
+      captures:
+        1: support.type.branch-name.gitcommit
+
+  change:
+    - match: (new file|modified|deleted)\s*(:)\s*(.*)
+      captures:
+        1: keyword.change-list.change.gitcommit
+        2: punctuation.separator.change.gitcommit
+        3: keyword.change-list.file.gitcommit
+
+  change-list:
+    - include: change-list-header
+    - include: change
+
+  change-list-header:
+    - match: '\bChanges to be committed\s*(:)'
+      scope: keyword.change-list.heading.gitcommit
+      captures:
+        1: punctuation.separator.change-list.gitcommit
+
+  comments:
+    - match: '#'
```

Comment:
Have you thought about this?

```
[foosection]
bar = baz#qux
one = two #three
```
Does `bar` have the value `baz` or does it have the value `baz#qux`?

Reply:
🤔 Hmm, currently it's matching as a value, ever since I changed [this match](https://github.com/sublimehq/Packages/pull/1126/files#diff-6a7ee7aed9d88d488d26be9dddef8e08R164) from `\S` to `\S+`.

![comment_or_value](https://user-images.githubusercontent.com/1250684/30358615-8921e0d0-97fa-11e7-8910-6314cdccc4fc.png)

I'm not sure if this is the correct behavior, or if it should match as a comment.

Reply:
I went ahead and tested this here (git version 2.11). With
```
[x]
    a = b#c
```
the output of
```
git config --get x.a
```
is
```
b
```

Reply:
Moreover,
```
"b#c".   ---> b#c
\"b#c\"  ---> "b
\"       ---> "